File: packages/create/ios/TvFocusableViewManager.m
#import "React/RCTViewManager.h"
#if TARGET_OS_TV
#import "React/RCTTVView.h"
#else
#import <React/RCTView.h>
#endif

#if TARGET_OS_TV
@interface TvFocusableView : RCTTVView
    @property (nonatomic, assign) NSDictionary *animatorOptions;
    @property (nonatomic, assign) BOOL *isTVSelectable;
@end
#else
@interface TvFocusableView : RCTView

@end
#endif

#if TARGET_OS_TV
@interface RCT_EXTERN_MODULE(TvFocusableViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(animatorOptions, NSDictionary)
RCT_EXPORT_VIEW_PROPERTY(isTVSelectable, BOOL)

RCT_EXTERN_METHOD(
    cmdFocus: (nonnull NSNumber *)node
)
RCT_EXTERN_METHOD(
    cmdBlur: (nonnull NSNumber *)node
)

@end
#endif

File: packages/create/ios/TvFocusableViewManager.swift
#if os(tvOS)
@objc(TvFocusableViewManager)
class TvFocusableViewManager: RCTViewManager {

  override func view() -> (TvFocusableView) {
    return TvFocusableView()
  }
    
    @objc func cmdBlur(_ node: NSNumber) -> Void {
        DispatchQueue.main.async {
            let component = self.bridge.uiManager.view(
              forReactTag: node
            ) as! TvFocusableView
            component.blur(animated: true)
        }
    }
    
    @objc func cmdFocus(_ node: NSNumber) -> Void {
        DispatchQueue.main.async {
            let component = self.bridge.uiManager.view(
              forReactTag: node
            ) as! TvFocusableView
            component.focus(animated: true)
        }
    }

    @objc override static func requiresMainQueueSetup() -> Bool {
      return false
    }
}
#endif